common/src/View/MapDocument.cpp

```diff
@@ -133,10 +118,79 @@
 #include <sstream>
 #include <string>
 #include <type_traits>
+#include <unordered_map>
 #include <vector>
 
 namespace TrenchBroom {
     namespace View {
+        template <typename N, typename L>
+        static std::optional<std::vector<std::pair<Model::Node*, Model::NodeContents>>> applyToNodeContents(const std::vector<N*>& nodes, L lambda) {
+            auto result = std::vector<std::pair<Model::Node*, Model::NodeContents>>{};
+            result.reserve(nodes.size());
+
+            bool success = true;
+            std::transform(std::begin(nodes), std::end(nodes), std::back_inserter(result), [&](auto* node) {
+                auto nodeContents = node->accept(kdl::overload(
```

Comment:
Took me some effort to figure out the type was `std::variant<Model::Entity, Model::Brush> nodeContents` so should probaby not use auto here.

Reply:
Agreed. df37fc16d